pkg/certificate/providers/vault/certificate_manager_test.go

```diff
@@ -114,99 +48,115 @@ var _ = Describe("Test client helpers", func() {
 			Expect(actual).To(Equal(expected))
 		})
 	})
+})
+
+func Test_New(t *testing.T) {
+	core, token, addr := mockVault(t)
+	defer core.Shutdown()
 
-	Context("Test Hashi Vault functions", func() {
-		cm := CertManager{
-			ca: rootCert,
-		}
-		cm.cache.Store(expiredCertCN, expiredCert)
-		cm.cache.Store(validCertCN, validCert)
-
-		getCachedCertificateCNs := func() []certificate.CommonName {
-			var commonNames []certificate.CommonName
-			cm.cache.Range(func(cnInterface interface{}, certInterface interface{}) bool {
-				cert := certInterface.(*certificate.Certificate)
-				commonNames = append(commonNames, cert.GetCommonName())
-				return true // continue the iteration
-			})
-			return commonNames
-		}
-
-		It("gets issuing CA public part", func() {
-			certSerialNum := certificate.SerialNumber(uuid.New().String())
-			expectedNumberOfCertsInCache := 2
-			Expect(len(getCachedCertificateCNs())).To(Equal(expectedNumberOfCertsInCache))
-			Expect(getCachedCertificateCNs()).To(ContainElement(certificate.CommonName("this.has.expired")))
-			Expect(getCachedCertificateCNs()).To(ContainElement(certificate.CommonName("valid.certificate")))
-			certBytes := uuid.New().String()
-			issue := func(certificate.CommonName, time.Duration) (*certificate.Certificate, error) {
-				cert := &certificate.Certificate{
-					IssuingCA:    pem.RootCertificate(certBytes),
-					SerialNumber: certSerialNum,
+	testCases := []struct {
+		description string
+		wantErr     bool
+	}{
+		{description: "no err", wantErr: false},
```

Comment:
Instead of err and no err, let's provide the description of why it fails, ie:

"errors when an invalid address is provided"